app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx

```diff
@@ -0,0 +1,343 @@
+import React, {
+	RefObject,
+	useCallback, useEffect, useRef, useState
+} from 'react';
+import { cx } from '@emotion/css';
+import { router } from '@forge/bridge';
+import Button, { ButtonGroup } from '@atlaskit/button';
+import InfoIcon from '@atlaskit/icon/glyph/info';
+import CopyIcon from '@atlaskit/icon/glyph/copy';
+import OpenIcon from '@atlaskit/icon/glyph/open';
+import Tooltip from '@atlaskit/tooltip';
+import PeopleGroup from '@atlaskit/icon/glyph/people-group';
+import { useParams } from 'react-router';
+import Spinner from '@atlaskit/spinner';
+import {
+	connectionFlowContainer,
+	connectionFlowInnerContainer,
+	orderedList,
+	orderedListItem
+} from '../../GlobalStyles.styles';
+import {
+	jenkinsSetupContainer,
+	jenkinsSetupContent,
+	jenkinsSetupCopyButtonContainer,
+	jenkinsSetupCopyContainer,
+	jenkinsSetupHeader,
+	jenkinsSetupListItem,
+	jenkinsSetupOrderedList,
+	jenkinsSetupServerName,
+	jenkinsSetUpInfoPanel,
+	jenkinsSetUpInfoPanelContentContainer,
+	jenkinsSetUpCopyHiddenContent,
+	loadingContainer
+} from './JenkinsSetup.styles';
+import { getJenkinsServerWithSecret } from '../../api/getJenkinsServerWithSecret';
+import { ParamTypes } from '../ConnectJenkins/ConnectJenkins/ConnectJenkins';
+import { serverNameFormOuterContainer } from '../ServerNameForm/ServerNameForm.styles';
+import { InProductHelpAction, InProductHelpActionType } from '../InProductHelpDrawer/InProductHelpAction';
+import { CopiedToClipboard } from '../CopiedToClipboard/CopiedToClipboard';
+import { ConnectionFlowHeader } from '../ConnectionWizard/ConnectionFlowHeader';
+import { SecretTokenContent, WebhookGuideContent } from '../CopiedToClipboard/CopyToClipboardContent';
+import { getWebhookUrl } from '../../common/util/jenkinsConnectionsUtils';
+import { fetchSiteName } from '../../api/fetchGlobalPageUrl';
+
+type CopyProps = {
+	handleCopyToClipboard: (copyRef: React.RefObject<HTMLDivElement>) => Promise<void> | void;
+	testId?: string
+};
+
+const CopyButton = ({
+	handleCopyToClipboard,
+	copyRef,
+	testId
+}: CopyProps & { copyRef: React.RefObject<HTMLDivElement> }): JSX.Element => {
+	const [isCopied, setIsCopied] = useState(false);
+
+	useEffect(() => {
+		let timeoutId: NodeJS.Timeout;
+		if (isCopied) {
+			timeoutId = setTimeout(() => {
+				setIsCopied(false);
+			}, 2000);
+		}
+		return () => {
+			if (timeoutId) {
+				clearTimeout(timeoutId);
+			}
+		};
+	}, [isCopied]);
+
+	return (
+		<div className={cx(jenkinsSetupCopyButtonContainer)} >
+			<Button
+				iconBefore={<CopyIcon label="Copy" size="medium" />}
+				onClick={() => {
+					if (copyRef && copyRef.current) {
+						handleCopyToClipboard(copyRef);
+						setIsCopied(true);
+					}
+				}}
+				testId={testId}
+			>
+				Copy
+			</Button>
+
+			{isCopied ? <CopiedToClipboard /> : null}
```

Comment:
/nit can we simplify this? seems hard work to have an else null
`{isCopied && <CopiedToClipboard />}`